src/cargo/ops/cargo_compile.rs

```diff
@@ -582,6 +590,82 @@ pub fn create_bcx<'a, 'cfg>(
         }
     }
 
+    if rustdoc_scrape_examples && units.len() > 0 {
+        let mut args = Vec::new();
+
+        let get_repository = || -> Option<String> {
+            let repository = units[0].pkg.manifest().metadata().repository.as_ref()?;
+            let repo_url = Url::parse(repository).ok()?;
+            let domain = repo_url.domain()?;
+
+            match domain {
+                "github.com" => {
+                    // TODO(wcrichto): is there a canonical way to get the current commit?
```

Comment:
git rev-parse HEAD; not sure if cargo already has a wrapper around that.

Note that this whole segment of code won't be necessary if rustdoc generates the sources for examples instead (which I think it should; linking to the repository also seems useful but is orthogonal and needs its own RFC since rustdoc should do it consistently for everything).

Reply:
No longer an issue since we're generating examples.